build/Dockerfile

```diff
@@ -1,46 +1,289 @@
+
+
+############################################# Base image for Openshift with NGINX Plus #############################################
+FROM openshift-base AS openshift-plus
+
+ENV NGINX_PLUS_VERSION 23-1.el8.ngx
+ARG IC_VERSION
+
+RUN rpm --import https://nginx.org/keys/nginx_signing.key \
+	&& echo "[nginx-plus]" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "name=nginx-plus repo" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "baseurl=https://plus-pkgs.nginx.com/centos/8/\$basearch/" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientcert=/etc/ssl/nginx/nginx-repo.crt" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientkey=/etc/ssl/nginx/nginx-repo.key" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "gpgcheck=1" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "enabled=1" >> /etc/yum.repos.d/nginx-plus-8.repo
+
+# Install NGINX Plus
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	yum install -y ca-certificates nginx-plus-${NGINX_PLUS_VERSION}
+
+
+############################################# Build image for Opentracing #############################################
+
+FROM debian as opentracing-builder
+
+ARG NGINX_VERSION
+ARG NGINX_OPENTRACING=0.10.0
+ARG OPENTRACING_VERSION=1.6.0
+
+RUN apt-get update && apt-get install -y -q --fix-missing --no-install-recommends \
+	autoconf \
+	automake \
+	build-essential \
+	cmake \
+	git \
+	libcurl4-openssl-dev \
+	libcurl4-openssl-dev \
+	libgeoip-dev \
+	liblmdb-dev \
+	libpcre3-dev \
+	libprotobuf-dev \
+	libssl-dev \
+	libtool \
+	libxml2-dev \
+	libyajl-dev \
+	pkgconf \
+	protobuf-compiler \
+	zlib1g-dev
+
+# Get nginx to build against
+RUN curl -sS -O -L http://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz && \
+	tar zxvf nginx-${NGINX_VERSION}.tar.gz && rm -f nginx-${NGINX_VERSION}.tar.gz
+
+RUN git clone --branch v${NGINX_OPENTRACING} https://github.com/opentracing-contrib/nginx-opentracing.git
+RUN git clone --branch v${OPENTRACING_VERSION} https://github.com/opentracing/opentracing-cpp.git
+
+RUN mkdir -p opentracing-cpp/.build && cd opentracing-cpp/.build && \
+	cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DBUILD_STATIC_LIBS=OFF -DBUILD_MOCKTRACER=OFF -DENABLE_LINTING=OFF .. && \
+	make && make install
+
+RUN cd nginx-${NGINX_VERSION} && \
+	./configure \
+	--with-compat --with-debug \
+	--add-dynamic-module=/nginx-opentracing/opentracing && \
+	make modules
+
+FROM debian AS tracer-downloader
+
+ARG JAEGER_VERSION=v0.4.2
+
+RUN set -x \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates apt-transport-https wget \
+	&& wget https://github.com/jaegertracing/jaeger-client-cpp/releases/download/${JAEGER_VERSION}/libjaegertracing_plugin.linux_amd64.so -O /usr/local/lib/libjaegertracing_plugin.so
+
+
+FROM debian as opentracing
```

Comment:
could  you add a comment here for consistency and visibility?

Reply:
added